lxd/device/config/devices_sort.go

```diff
@@ -49,6 +49,15 @@ func (devices DevicesSortable) Less(i, j int) bool {
 	// Start disk devices in path order.
 	if a.Config["type"] == "disk" && b.Config["type"] == "disk" {
 		if a.Config["path"] != b.Config["path"] {
+			// The root device always goes first.
+			if a.Config["path"] == "/" {
+				return true
+			}
+
+			if b.Config["path"] == "/" {
+				return false
+			}
+
 			return a.Config["path"] < b.Config["path"]
```

Comment:
@monstermunchkin did the previous one not work because of disk devices with no `path`?

Reply:
Yes

Reply:
Cool, nice we fixed that then.